pkg/kubelet/oom_watcher.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2015 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubelet
+
+import (
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/record"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/cadvisor"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"github.com/google/cadvisor/events"
+	cadvisorApi "github.com/google/cadvisor/info/v1"
+)
+
+type OOMWatcher interface {
+	RecordSysOOMs(node *api.Node) error
+}
+
+type realOOMWatcher struct {
+	cadvisor cadvisor.Interface
+	recorder record.EventRecorder
+}
+
+func NewOOMWatcher(cadvisor cadvisor.Interface, recorder record.EventRecorder) OOMWatcher {
+	return &realOOMWatcher{
+		cadvisor: cadvisor,
+		recorder: recorder,
+	}
+}
+
+const systemOOMEvent = "SystemOOM"
+
+func (self *realOOMWatcher) RecordSysOOMs(node *api.Node) error {
```

Comment:
nit: use `ow` instead of `self` since it is the Go style apparently. We migrated all of Kubelet to it recently.